docs/02-app/02-api-reference/04-functions/use-router.mdx

```diff
@@ -41,8 +41,8 @@ export default function Page() {
 
 ## `useRouter()`
 
-- `router.push(href: string, { scroll: boolean })`: Perform a client-side navigation to the provided route. Adds a new entry into the [browser’s history](https://developer.mozilla.org/docs/Web/API/History_API) stack.
-- `router.replace(href: string, { scroll: boolean })`: Perform a client-side navigation to the provided route without adding a new entry into the [browser’s history stack](https://developer.mozilla.org/docs/Web/API/History_API).
+- `router.push(href: string, { scroll: boolean })`: Perform a client-side navigation to the provided route. Adds a new entry into the [browser’s history](https://developer.mozilla.org/docs/Web/API/History_API) stack. Note that you must not send untrusted or unsanitized URLs to `router.push`, as this can open your site to cross-site scripting (XSS) vulnerabilities. For example, `javascript:` URLs sent to `router.push` will be executed in the context of your page.
+- `router.replace(href: string, { scroll: boolean })`: Perform a client-side navigation to the provided route without adding a new entry into the [browser’s history stack](https://developer.mozilla.org/docs/Web/API/History_API). Note that you must not send untrusted or unsanitized URLs to `router.replace`, as this can open your site to cross-site scripting (XSS) vulnerabilities. For example, `javascript:` URLs sent to `router.replace` will be executed in the context of your page.
```

Comment:
I would break this out into a separate note so it's not duplicated, also this kinda feels like a generic warning you shouldn't pass untrusted user input to any API really.

Reply:
Thanks for the comment! I'll pull the note out so that it's not duplicated.

As for this being a generic warning that applies to any API, I think I respectfully disagree. In my experience, there are a lot of APIs that are designed to accept untrusted user input. For example, `React.createElement` escapes strings to avoid XSS specifically so that you can display untrusted user input. As another example, the browser API that is closest to `router.push`, `history.pushState`, is not susceptible to XSS URLs and can accept untrusted user input, as far as I understand.

Also, as I noted in the original bug report, with minimal searching I found over 1,000 public projects on Github that are probably vulnerable to this problem, and that is certainly an undercount, as I only looked for ones that called their user variable `callbackUrl`. So even if this is generic advice, it seems like developers are not understanding that it needs to be followed in this particular instance.

Reply:
I pulled the note out and put it in the "Good to know" section in 1024026 so that it isn't duplicated. Thanks for the review!